### ISSUE TITLE
Missing state prop in custom day component

### ISSUE THREAD

**Issue by CauchR6** [wix/react-native-calendars#1393]
I tried implementing a custom day component to override the marking type in the calendar. I followed the documentation related to this, but I receive different parameters than what is enumerated in the documentation. See below for more info.

## Description

The problem is the dayComponent override does not give marking, state and other props mentionned like onDayPress.

## Expected Behavior

I expected the signature to correspond to the doc. If it is an issue related to removed props in later versions, then I would like to know how I can access those props to disable some dates in custom day components. 
Expected behavior (taken from v1.403.0): 
![Screen Shot 2021-01-15 at 10 56 32 AM](https://user-images.githubusercontent.com/77501175/104749294-a1cbf380-5720-11eb-9e1e-43f181349322.png)

## Observed Behavior
See this screenshot for current behavior (on v1.1129.0):
![Screen Shot 2021-01-15 at 10 57 50 AM](https://user-images.githubusercontent.com/77501175/104749246-937dd780-5720-11eb-855c-a4f61f316161.png)

This is an example entry of what I receive from dayComponent override.
```
{
  "accessibilityLabel": " Friday 8 January 2021 ", 
  "children": 8, 
  "date": {"dateString": "2021-01-08", "day": 8, "month": 1, "timestamp": 1610064000000, "year": 2021},
  "testID": "native.calendar.SELECT_DATE_SLOT-2021-01-08"
}
```
Notice there is no marking or state entry. Those are then undefined.

## Environment

* react-native-calendars is running version '1.1129.0' which is at the time writing this, the latest release on NPM.
**EDIT: I tested the version 1.403 of the lib following** [this discussion](https://github.com/wix/react-native-calendars/issues/1373) and it works. It also has marking and the other props missing. Maybe those were changed in later versions and the doc was not updated? 
* react-native is on version '0.63.4'.

Also specify:

1. Phone/emulator/simulator & version: ios simulator on latest IOS version.

## Reproducible Demo

```
const renderDayComponent = ({
    date,
    state,
  }: {
    date: CalendarDate
    state: any
  }) => {
    console.log(state) // State is undefined on all entries.
    const isDisabled = state === 'disabled' **// always falsy**
    const textStyle = isDisabled ? styles.disabledText : styles.defaultDateText

    return (
      <View style={mark[date.dateString]?.customStyles.container}>
        <TouchableOpacity onPress={() => onPressDate(date)}>
          <Text style={[styles.center, textStyle]}>{date.day}</Text>
          <View style={styles.markingRow}>
            {checkCompliance(date) ? (
              <CheckmarkIcon
                size={Theme.metrics.shape.size.small}
                color={Theme.colors.primaryDark}
              />
            ) : (
              <Text style={styles.notCompliantText}>x</Text>
            )}
          </View>
        </TouchableOpacity>
      </View>
    )
  }

// ...
<Calendar
            ref={calendarRef}
            current={selectedDate}
            onDayPress={onChangeDate}
            style={styles.calendar}
            minDate={startDate.atStartOfMonth().toDayFormat()}
            maxDate={lastCalendarDate}
            markedDates={mark}
            renderHeader={(date: Date) => Header(date)}
            onMonthChange={({ dateString }: { dateString: string }) =>
              handleMonthChange(dateString)
            }
            dayComponent={renderDayComponent}
            enableSwipeMonths
            disableMonthChange
            disableAllTouchEventsForDisabledDays
          />
```

**Comment by jhogervorst** [comment]
I’m having a similar problem, missing the `onPress` prop in my custom day component.

Some trial-and-error showed that this issue started appearing since version **1.830.0** on NPM. This seems to be caused by one of the commits from Dec 10, 2020.

As workaround, I installed the previous version **1.829.0** using `npm install react-native-calendars@1.829.0`.

**Comment by BossBele** [comment]
Same issue with state prop in custom day component --- latest version

**Comment by SergeySlutskiyGowombat** [comment]
I’m having a similar problem. found that the lib provide rest of props using propTypes

**Comment by YarmolenkoD** [comment]
It helped me https://github.com/wix/react-native-calendars/issues/1373#issuecomment-746625048

**Comment by jetBn** [comment]
how to solve it ?

**Comment by SergeySlutskiyGowombat** [comment]
> how to solve it ?

You can add PropTypes to your custom day component.


> I used props from the PeriodDay:
> static propTypes = {
> state: PropTypes.oneOf(['selected', 'disabled', 'today', '']), 
> marking: PropTypes.any,
> theme: PropTypes.object,
> onPress: PropTypes.func,
> onLongPress: PropTypes.func,
> date: PropTypes.object
> };
 it helps me

**Comment by PatrickBokhove** [comment]
I'm experiencing the same issue since i did a full new `npm install`today. `marking` and `onPress` seem to be undefined in my memoized dayComponent.

**Comment by patradanai** [comment]
> DayComponent.propTypes = {
  state: PropTypes.oneOf(['selected', 'disabled', 'today', '']),
  marking: PropTypes.any,
  theme: PropTypes.object,
  onPress: PropTypes.func,
  onLongPress: PropTypes.func,
  date: PropTypes.object,
};

I fiexd it with propsTypes 

According to Lib Code 

`function extractComponentProps(component, props, ignoreProps) {
  const componentPropTypes = component.propTypes; <<<<< ---------------- This Called propTypes of Component
  if (componentPropTypes) {
    const keys = Object.keys(componentPropTypes);
    const componentProps = _.chain(props)
      .pickBy((_value, key) => _.includes(keys, key))
      .omit(ignoreProps)
      .value();
    return componentProps;
  }
  return {};
}`

**Comment by quizzy** [comment]
This patch fixed it for me #1407
